src/resolvers/registries/npm-resolver.js

```diff
@@ -107,8 +107,19 @@ export default class NpmResolver extends RegistryResolver {
 
     const files = await this.config.getCache('cachedPackages', async (): Promise<Array<string>> => {
       const files = await fs.readdir(cacheFolder);
-      const validFiles = [];
+      const scoped_files = {};
+      for (const scope of files) {
+        // no hidden files
+        if (scope[0] === '.') {
+          continue;
+        }
+
+        if (!await fs.exists(path.join(cacheFolder, scope, METADATA_FILENAME))) {
```

Comment:
Using async exists is discouraged by Node folks since the file may not be there when you actually want to use it. I guess that's not very different from what is previously done but this now adds a lot more I/O during offline-cache init.